File: src/main/kotlin/com/huawei/cangjie/psi/stubs/elements/CjStubElementTypes.java
package com.huawei.cangjie.psi.stubs.elements;

import com.huawei.cangjie.psi.*;
import com.intellij.psi.tree.TokenSet;

public interface CjStubElementTypes {


    CjEnumEntryElementType ENUM_ENTRY = new CjEnumEntryElementType("ENUM_ENTRY");

    CjClassElementType CLASS = new CjClassElementType("CLASS");
    CjFunctionElementType FUNCTION = new CjFunctionElementType("FUNC");

    CjVariableElementType VARIABLE = new CjVariableElementType("VARIABLE");
    CjPropertyElementType PROPERTY = new CjPropertyElementType("PROPERTY");
    CjPropertyAccessorElementType PROPERTY_ACCESSOR = new CjPropertyAccessorElementType("PROPERTY_ACCESSOR");
//    CjBackingFieldElementType BACKING_FIELD = new CjBackingFieldElementType("BACKING_FIELD");
    CjTypeAliasElementType TYPEALIAS = new CjTypeAliasElementType("TYPEALIAS");

//    CjObjectElementType OBJECT_DECLARATION = new CjObjectElementType("OBJECT_DECLARATION");
    CjPlaceHolderStubElementType<CjClassInitializer> CLASS_INITIALIZER =
            new CjPlaceHolderStubElementType<>("CLASS_INITIALIZER", CjClassInitializer.class);
    CjSecondaryConstructorElementType SECONDARY_CONSTRUCTOR =
            new CjSecondaryConstructorElementType("SECONDARY_CONSTRUCTOR");
    CjPrimaryConstructorElementType PRIMARY_CONSTRUCTOR =
            new CjPrimaryConstructorElementType("PRIMARY_CONSTRUCTOR");

    CjParameterElementType VALUE_PARAMETER = new CjParameterElementType("VALUE_PARAMETER");
    CjPlaceHolderStubElementType<CjParameterList> VALUE_PARAMETER_LIST =
            new CjPlaceHolderStubElementType<>("VALUE_PARAMETER_LIST", CjParameterList.class);

    CjTypeParameterElementType TYPE_PARAMETER = new CjTypeParameterElementType("TYPE_PARAMETER");
    CjPlaceHolderStubElementType<CjTypeParameterList> TYPE_PARAMETER_LIST =
            new CjPlaceHolderStubElementType<>("TYPE_PARAMETER_LIST", CjTypeParameterList.class);

    CjAnnotationEntryElementType ANNOTATION_ENTRY = new CjAnnotationEntryElementType("ANNOTATION_ENTRY");
    CjPlaceHolderStubElementType<CjAnnotation> ANNOTATION =
            new CjPlaceHolderStubElementType<>("ANNOTATION", CjAnnotation.class);

//    CjAnnotationUseSiteTargetElementType ANNOTATION_TARGET = new CjAnnotationUseSiteTargetElementType("ANNOTATION_TARGET");

    CjPlaceHolderStubElementType<CjClassBody> CLASS_BODY =
            new CjPlaceHolderStubElementType<>("CLASS_BODY", CjClassBody.class);

    CjPlaceHolderStubElementType<CjImportList> IMPORT_LIST =
            new CjPlaceHolderStubElementType<>("IMPORT_LIST", CjImportList.class);

//    CjPlaceHolderStubElementType<CjFileAnnotationList> FILE_ANNOTATION_LIST =
//            new CjPlaceHolderStubElementType<>("FILE_ANNOTATION_LIST", CjFileAnnotationList.class);

    CjImportDirectiveElementType IMPORT_DIRECTIVE = new CjImportDirectiveElementType("IMPORT_DIRECTIVE");

    CjImportAliasElementType IMPORT_ALIAS = new CjImportAliasElementType("IMPORT_ALIAS");

//    CjPlaceHolderStubElementType<CjPackageDirective> PACKAGE_DIRECTIVE =
//            new CjPlaceHolderStubElementType<>("PACKAGE_DIRECTIVE", CjPackageDirective.class);
CjPackageDirectiveElementType  PACKAGE_DIRECTIVE =
        new CjPackageDirectiveElementType("PACKAGE_DIRECTIVE" );

    CjModifierListElementType<CjDeclarationModifierList> MODIFIER_LIST =
            new CjModifierListElementType<>("MODIFIER_LIST", CjDeclarationModifierList.class);

    CjPlaceHolderStubElementType<CjTypeConstraintList> TYPE_CONSTRAINT_LIST =
            new CjPlaceHolderStubElementType<>("TYPE_CONSTRAINT_LIST", CjTypeConstraintList.class);

    CjPlaceHolderStubElementType<CjTypeConstraint> TYPE_CONSTRAINT =
            new CjPlaceHolderStubElementType<>("TYPE_CONSTRAINT", CjTypeConstraint.class);

//    CjPlaceHolderStubElementType<CjNullableType> NULLABLE_TYPE =
//            new CjPlaceHolderStubElementType<>("NULLABLE_TYPE", CjNullableType.class);

//    CjPlaceHolderStubElementType<CjIntersectionType> INTERSECTION_TYPE =
//            new CjPlaceHolderStubElementType<>("INTERSECTION_TYPE", CjIntersectionType.class);
CjPlaceHolderStubElementType<CjOptionType> OPTIONAL_TYPE =
        new CjPlaceHolderStubElementType<>("OPTIONAL_TYPE", CjOptionType.class);

    CjPlaceHolderStubElementType<CjTypeReference> TYPE_REFERENCE =
            new CjPlaceHolderStubElementType<>("TYPE_REFERENCE", CjTypeReference.class);
    CjBasicTypeElementType BASIC_TYPE = new CjBasicTypeElementType("BASIC_TYPE");

    CjUserTypeElementType USER_TYPE = new CjUserTypeElementType("USER_TYPE");
//    CjPlaceHolderStubElementType<CjDynamicType> DYNAMIC_TYPE =
//            new CjPlaceHolderStubElementType<>("DYNAMIC_TYPE", CjDynamicType.class);

    CjPlaceHolderStubElementType<CjFunctionType> FUNCTION_TYPE =
            new CjPlaceHolderStubElementType<>("FUNCTION_TYPE", CjFunctionType.class);


    CjTypeProjectionElementType TYPE_PROJECTION = new CjTypeProjectionElementType("TYPE_PROJECTION");

    CjPlaceHolderStubElementType<CjFunctionTypeReceiver> FUNCTION_TYPE_RECEIVER =
            new CjPlaceHolderStubElementType<>("FUNCTION_TYPE_RECEIVER", CjFunctionTypeReceiver.class);

    CjNameReferenceExpressionElementType REFERENCE_EXPRESSION = new CjNameReferenceExpressionElementType("REFERENCE_EXPRESSION");
    CjDotQualifiedExpressionElementType DOT_QUALIFIED_EXPRESSION = new CjDotQualifiedExpressionElementType("DOT_QUALIFIED_EXPRESSION");
//    CjEnumEntrySuperClassReferenceExpressionElementType
//            ENUM_ENTRY_SUPERCLASS_REFERENCE_EXPRESSION =
//            new CjEnumEntrySuperClassReferenceExpressionElementType("ENUM_ENTRY_SUPERCLASS_REFERENCE_EXPRESSION");
    CjPlaceHolderStubElementType<CjTypeArgumentList> TYPE_ARGUMENT_LIST =
            new CjPlaceHolderStubElementType<>("TYPE_ARGUMENT_LIST", CjTypeArgumentList.class);

    CjPlaceHolderStubElementType<CjValueArgumentList> VALUE_ARGUMENT_LIST =
            new CjValueArgumentListElementType("VALUE_ARGUMENT_LIST");

    CjValueArgumentElementType<CjValueArgument> VALUE_ARGUMENT =
            new CjValueArgumentElementType<>("VALUE_ARGUMENT", CjValueArgument.class);

//    CjPlaceHolderStubElementType<CjContractEffectList> CONTRACT_EFFECT_LIST =
//            new CjContractEffectListElementType("CONTRACT_EFFECT_LIST");

//    CjContractEffectElementType CONTRACT_EFFECT =
//            new CjContractEffectElementType("CONTRACT_EFFECT", CjContractEffect.class);

    CjValueArgumentElementType<CjLambdaArgument> LAMBDA_ARGUMENT =
            new CjValueArgumentElementType<>("LAMBDA_ARGUMENT", CjLambdaArgument.class);

    CjPlaceHolderStubElementType<CjValueArgumentName> VALUE_ARGUMENT_NAME =
            new CjPlaceHolderStubElementType<>("VALUE_ARGUMENT_NAME", CjValueArgumentName.class);

    CjPlaceHolderStubElementType<CjSuperTypeList> SUPER_TYPE_LIST =
            new CjPlaceHolderStubElementType<>("SUPER_TYPE_LIST", CjSuperTypeList.class);

//    CjPlaceHolderStubElementType<CjInitializerList> INITIALIZER_LIST =
//            new CjPlaceHolderStubElementType<>("INITIALIZER_LIST", CjInitializerList.class);

//    CjPlaceHolderStubElementType<CjDelegatedSuperTypeEntry> DELEGATED_SUPER_TYPE_ENTRY =
//            new CjPlaceHolderStubElementType<>("DELEGATED_SUPER_TYPE_ENTRY", CjDelegatedSuperTypeEntry.class);

    CjPlaceHolderStubElementType<CjSuperTypeCallEntry> SUPER_TYPE_CALL_ENTRY =
            new CjPlaceHolderStubElementType<>("SUPER_TYPE_CALL_ENTRY", CjSuperTypeCallEntry.class);
    CjPlaceHolderStubElementType<CjSuperTypeEntry> SUPER_TYPE_ENTRY =
            new CjPlaceHolderStubElementType<>("SUPER_TYPE_ENTRY", CjSuperTypeEntry.class);
    CjPlaceHolderStubElementType<CjConstructorCalleeExpression> CONSTRUCTOR_CALLEE =
            new CjPlaceHolderStubElementType<>("CONSTRUCTOR_CALLEE", CjConstructorCalleeExpression.class);

    CjContextReceiverElementType CONTEXT_RECEIVER = new CjContextReceiverElementType("CONTEXT_RECEIVER");
    CjPlaceHolderStubElementType<CjContextReceiverList> CONTEXT_RECEIVER_LIST =
            new CjPlaceHolderStubElementType<>("CONTEXT_RECEIVER_LIST", CjContextReceiverList.class);

//    CjConstantExpressionElementType NULL                = new CjConstantExpressionElementType("NULL");
    CjConstantExpressionElementType BOOLEAN_CONSTANT    = new CjConstantExpressionElementType("BOOLEAN_CONSTANT");
    CjConstantExpressionElementType FLOAT_CONSTANT      = new CjConstantExpressionElementType("FLOAT_CONSTANT");
//    CjConstantExpressionElementType CHARACTER_CONSTANT  = new CjConstantExpressionElementType("CHARACTER_CONSTANT");
    CjConstantExpressionElementType INTEGER_CONSTANT    = new CjConstantExpressionElementType("INTEGER_CONSTANT");
//    CjClassLiteralExpressionElementType CLASS_LITERAL_EXPRESSION = new CjClassLiteralExpressionElementType("CLASS_LITERAL_EXPRESSION");
CjConstantExpressionElementType RUNE_CONSTANT = new CjConstantExpressionElementType("RUNE_CONSTANT");


    CjCollectionLiteralExpressionElementType COLLECTION_LITERAL_EXPRESSION = new CjCollectionLiteralExpressionElementType("COLLECTION_LITERAL_EXPRESSION");

    CjPlaceHolderStubElementType<CjStringTemplateExpression> STRING_TEMPLATE =
            new CjStringTemplateExpressionElementType("STRING_TEMPLATE");

    CjPlaceHolderWithTextStubElementType<CjBlockStringTemplateEntry> LONG_STRING_TEMPLATE_ENTRY =
            new CjPlaceHolderWithTextStubElementType<>("LONG_STRING_TEMPLATE_ENTRY", CjBlockStringTemplateEntry.class);

    CjPlaceHolderWithTextStubElementType<CjSimpleNameStringTemplateEntry> SHORT_STRING_TEMPLATE_ENTRY =
            new CjPlaceHolderWithTextStubElementType<>("SHORT_STRING_TEMPLATE_ENTRY", CjSimpleNameStringTemplateEntry.class);

    CjPlaceHolderWithTextStubElementType<CjLiteralStringTemplateEntry> LITERAL_STRING_TEMPLATE_ENTRY =
            new CjPlaceHolderWithTextStubElementType<>("LITERAL_STRING_TEMPLATE_ENTRY", CjLiteralStringTemplateEntry.class);

    CjPlaceHolderWithTextStubElementType<CjEscapeStringTemplateEntry> ESCAPE_STRING_TEMPLATE_ENTRY =
            new CjPlaceHolderWithTextStubElementType<>("ESCAPE_STRING_TEMPLATE_ENTRY", CjEscapeStringTemplateEntry.class);





    CjConstantExpressionElementType UNIT_CONSTANT = new CjConstantExpressionElementType("UNIT_CONSTANT");
    CjAnnotationEntryElementType MACRO_EXPRESSION = new CjAnnotationEntryElementType("MACRO_EXPRESSION");


    //    CjPlaceHolderStubElementType<CjAnnotation> ANNOTATION =
//            new CjPlaceHolderStubElementType<>("ANNOTATION", CjAnnotation.class);

    CjConstantExpressionElementType CHARACTER_BYTE_CONSTANT = new CjConstantExpressionElementType("CHARACTER_BYTE_CONSTANT");







    CjInterfaceElementType INTERFACE = new CjInterfaceElementType("INTERFACE");

    CjStructElementType STRUCT = new CjStructElementType("STRUCT");



    CjMacroElementType MACRO = new CjMacroElementType("MACRO");
    CjForeignDirectiveElementType FOREIGN = new CjForeignDirectiveElementType("FOREIGN");
    CjPlaceHolderStubElementType<CjForeignBody> FOREIGN_BODY =
            new CjPlaceHolderStubElementType<>("FOREIGN_BODY", CjForeignBody.class);


    CjEnumElementType ENUM = new CjEnumElementType("ENUM");
    CjExtendElementType EXTEND = new CjExtendElementType("EXTEND");

    //    CjTupleTypeElementType TUPLE_TYPE = new CjTupleTypeElementType("TUPLE_TYPE");
    CjPlaceHolderStubElementType<CjTupleType> TUPLE_TYPE = new CjPlaceHolderStubElementType<>("TUPLE_TYPE", CjTupleType.class);
    CjPlaceHolderStubElementType<CjParenthesizedType> PARENTHESIZED_TYPE = new CjPlaceHolderStubElementType<>("PARENTHESIZED_TYPE", CjParenthesizedType.class);


    CjPlaceHolderStubElementType<CjEnumBody> ENUM_BODY =
            new CjPlaceHolderStubElementType<>("ENUM_BODY", CjEnumBody.class);

    CjPlaceHolderStubElementType<CjPropertyBody> PROPERTY_BODY =
            new CjPlaceHolderStubElementType<>("PROPERTY_BODY", CjPropertyBody.class);
    CjPlaceHolderStubElementType<CjPropertyGet> PROPERTY_GET =
            new CjPlaceHolderStubElementType<>("PROPERTY_GET", CjPropertyGet.class);

    CjPlaceHolderStubElementType<CjPropertySet> PROPERTY_SET =
            new CjPlaceHolderStubElementType<>("PROPERTY_SET", CjPropertySet.class);


    CjPlaceHolderStubElementType<CjEnumEntryTypeEntry> TYPE_LIST =
            new CjPlaceHolderStubElementType<>("TYPE_LIST", CjEnumEntryTypeEntry.class);


//    CjImportDirectiveItemElementType IMPORT_DIRECTIVE_ITEM = new CjImportDirectiveItemElementType("IMPORT_DIRECTIVE_ITEM");


    CjMainFunctionElementType MAIN_FUNC = new CjMainFunctionElementType("MAIN_FUNC");
    CjClassInitElementType CLASS_INIT = new CjClassInitElementType("CLASS_INIT");
    CjClassInitElementType CLASS_MAIN_INIT = new CjClassInitElementType("CLASS_MAIN_INIT");
    CjClassInitElementType CLASS_TILDE_INIT = new CjClassInitElementType("CLASS_TILDE_INIT");

//    CjBasicTypeElementType BASIC_TYPE = new CjBasicTypeElementType("BASIC_TYPE");










}

File: src/main/kotlin/com/huawei/cangjie/psi/stubs/elements/CjStubElementTypes.kt
//package com.huawei.cangjie.psi.stubs.elements
//
//import com.huawei.cangjie.psi.*
//import com.intellij.psi.tree.TokenSet
//
//interface CjStubElementTypes {
//
//
//    val CLASS: CjClassElementType = CjClassElementType("CLASS")
//
//    @JvmField
//    val FUNCTION: CjFunctionElementType = CjFunctionElementType("FUNC")
//
//    @JvmField
//    val VARIABLE: CjVariableElementType = CjVariableElementType("VARIABLE")
//
//    @JvmField
//    val PROPERTY: CjPropertyElementType = CjPropertyElementType("PROPERTY")
//
//    @JvmField
//    val PROPERTY_ACCESSOR: CjPropertyAccessorElementType = CjPropertyAccessorElementType("PROPERTY_ACCESSOR")
//
//    //    CjBackingFieldElementType BACKING_FIELD = new CjBackingFieldElementType("BACKING_FIELD");
//    @JvmField
//    val TYPEALIAS: CjTypeAliasElementType = CjTypeAliasElementType("TYPEALIAS")
//
//    @JvmField
//    val ENUM_ENTRY: CjClassElementType = CjClassElementType("ENUM_ENTRY")
//
//    //    CjObjectElementType OBJECT_DECLARATION = new CjObjectElementType("OBJECT_DECLARATION");
//    @JvmField
//    val CLASS_INITIALIZER: CjPlaceHolderStubElementType<CjClassInitializer> = CjPlaceHolderStubElementType(
//        "CLASS_INITIALIZER",
//        CjClassInitializer::class.java
//    )
//
//    @JvmField
//    val SECONDARY_CONSTRUCTOR: CjSecondaryConstructorElementType =
//        CjSecondaryConstructorElementType("SECONDARY_CONSTRUCTOR")
//
//    @JvmField
//    val PRIMARY_CONSTRUCTOR: CjPrimaryConstructorElementType = CjPrimaryConstructorElementType("PRIMARY_CONSTRUCTOR")
//
//    @JvmField
//    val VALUE_PARAMETER: CjParameterElementType = CjParameterElementType("VALUE_PARAMETER")
//
//    @JvmField
//    val VALUE_PARAMETER_LIST: CjPlaceHolderStubElementType<CjParameterList> = CjPlaceHolderStubElementType(
//        "VALUE_PARAMETER_LIST",
//        CjParameterList::class.java
//    )
//
//    @JvmField
//    val TYPE_PARAMETER: CjTypeParameterElementType = CjTypeParameterElementType("TYPE_PARAMETER")
//
//    @JvmField
//    val TYPE_PARAMETER_LIST: CjPlaceHolderStubElementType<CjTypeParameterList> = CjPlaceHolderStubElementType(
//        "TYPE_PARAMETER_LIST",
//        CjTypeParameterList::class.java
//    )
//
//    @JvmField
//    val ANNOTATION_ENTRY: CjAnnotationEntryElementType = CjAnnotationEntryElementType("ANNOTATION_ENTRY")
//
//    @JvmField
//    val ANNOTATION: CjPlaceHolderStubElementType<CjAnnotation> = CjPlaceHolderStubElementType(
//        "ANNOTATION",
//        CjAnnotation::class.java
//    )
//
//
//    //    CjAnnotationUseSiteTargetElementType ANNOTATION_TARGET = new CjAnnotationUseSiteTargetElementType("ANNOTATION_TARGET");
//    @JvmField
//    val CLASS_BODY: CjPlaceHolderStubElementType<CjClassBody> = CjPlaceHolderStubElementType(
//        "CLASS_BODY",
//        CjClassBody::class.java
//    )
//
//    @JvmField
//    val IMPORT_LIST: CjPlaceHolderStubElementType<CjImportList> = CjPlaceHolderStubElementType(
//        "IMPORT_LIST",
//        CjImportList::class.java
//    )
//
//
//    //    CjPlaceHolderStubElementType<CjFileAnnotationList> FILE_ANNOTATION_LIST =
//    //            new CjPlaceHolderStubElementType<>("FILE_ANNOTATION_LIST", CjFileAnnotationList.class);
//    @JvmField
//    val IMPORT_DIRECTIVE: CjImportDirectiveElementType = CjImportDirectiveElementType("IMPORT_DIRECTIVE")
//
//    @JvmField
//    val IMPORT_ALIAS: CjImportAliasElementType = CjImportAliasElementType("IMPORT_ALIAS")
//
//    @JvmField
//    val PACKAGE_DIRECTIVE: CjPlaceHolderStubElementType<CjPackageDirective> = CjPlaceHolderStubElementType(
//        "PACKAGE_DIRECTIVE",
//        CjPackageDirective::class.java
//    )
//
//    @JvmField
//    val MODIFIER_LIST: CjModifierListElementType<CjDeclarationModifierList> = CjModifierListElementType(
//        "MODIFIER_LIST",
//        CjDeclarationModifierList::class.java
//    )
//
//    @JvmField
//    val TYPE_CONSTRAINT_LIST: CjPlaceHolderStubElementType<CjTypeConstraintList> = CjPlaceHolderStubElementType(
//        "TYPE_CONSTRAINT_LIST",
//        CjTypeConstraintList::class.java
//    )
//
//    @JvmField
//    val TYPE_CONSTRAINT: CjPlaceHolderStubElementType<CjTypeConstraint> = CjPlaceHolderStubElementType(
//        "TYPE_CONSTRAINT",
//        CjTypeConstraint::class.java
//    )
//
//
//    //    CjPlaceHolderStubElementType<CjNullableType> NULLABLE_TYPE =
//    //            new CjPlaceHolderStubElementType<>("NULLABLE_TYPE", CjNullableType.class);
//    //    CjPlaceHolderStubElementType<CjIntersectionType> INTERSECTION_TYPE =
//    //            new CjPlaceHolderStubElementType<>("INTERSECTION_TYPE", CjIntersectionType.class);
//    @JvmField
//    val TYPE_REFERENCE: CjPlaceHolderStubElementType<CjTypeReference> = CjPlaceHolderStubElementType(
//        "TYPE_REFERENCE",
//        CjTypeReference::class.java
//    )
//
//    @JvmField
//    val USER_TYPE: CjUserTypeElementType = CjUserTypeElementType("USER_TYPE")
//
//
//    //    CjPlaceHolderStubElementType<CjDynamicType> DYNAMIC_TYPE =
//    //            new CjPlaceHolderStubElementType<>("DYNAMIC_TYPE", CjDynamicType.class);
//    @JvmField
//    val FUNCTION_TYPE: CjPlaceHolderStubElementType<CjFunctionType> = CjPlaceHolderStubElementType(
//        "FUNCTION_TYPE",
//        CjFunctionType::class.java
//    )
//
////    @JvmField
//@JvmStatic
//    val TYPE_CODE_FRAGMENT: CjTypeCodeFragmentType = CjTypeCodeFragmentType()
//
////    @JvmField
//@JvmStatic
//    val EXPRESSION_CODE_FRAGMENT: CjExpressionCodeFragmentType = CjExpressionCodeFragmentType()
//
////    @JvmField
//    @JvmStatic
//    val BLOCK_CODE_FRAGMENT: CjBlockCodeFragmentType = CjBlockCodeFragmentType()
//
//    @JvmField
//    val TYPE_PROJECTION: CjTypeProjectionElementType = CjTypeProjectionElementType("TYPE_PROJECTION")
//
//    @JvmField
//    val FUNCTION_TYPE_RECEIVER: CjPlaceHolderStubElementType<CjFunctionTypeReceiver> = CjPlaceHolderStubElementType(
//        "FUNCTION_TYPE_RECEIVER",
//        CjFunctionTypeReceiver::class.java
//    )
//
//    @JvmField
//    val REFERENCE_EXPRESSION: CjNameReferenceExpressionElementType =
//        CjNameReferenceExpressionElementType("REFERENCE_EXPRESSION")
//
//    @JvmField
//    val DOT_QUALIFIED_EXPRESSION: CjDotQualifiedExpressionElementType =
//        CjDotQualifiedExpressionElementType("DOT_QUALIFIED_EXPRESSION")
//
//    //    CjEnumEntrySuperClassReferenceExpressionElementType
//    //            ENUM_ENTRY_SUPERCLASS_REFERENCE_EXPRESSION =
//    //            new CjEnumEntrySuperClassReferenceExpressionElementType("ENUM_ENTRY_SUPERCLASS_REFERENCE_EXPRESSION");
//    @JvmField
//    val TYPE_ARGUMENT_LIST: CjPlaceHolderStubElementType<CjTypeArgumentList> = CjPlaceHolderStubElementType(
//        "TYPE_ARGUMENT_LIST",
//        CjTypeArgumentList::class.java
//    )
//
//    @JvmField
//    val VALUE_ARGUMENT_LIST: CjPlaceHolderStubElementType<CjValueArgumentList> =
//        CjValueArgumentListElementType("VALUE_ARGUMENT_LIST")
//
//    @JvmField
//    val VALUE_ARGUMENT: CjValueArgumentElementType<CjValueArgument> = CjValueArgumentElementType(
//        "VALUE_ARGUMENT",
//        CjValueArgument::class.java
//    )
//
//
//    //    CjPlaceHolderStubElementType<CjContractEffectList> CONTRACT_EFFECT_LIST =
//    //            new CjContractEffectListElementType("CONTRACT_EFFECT_LIST");
//    //    CjContractEffectElementType CONTRACT_EFFECT =
//    //            new CjContractEffectElementType("CONTRACT_EFFECT", CjContractEffect.class);
//    @JvmField
//    val LAMBDA_ARGUMENT: CjValueArgumentElementType<CjLambdaArgument> = CjValueArgumentElementType(
//        "LAMBDA_ARGUMENT",
//        CjLambdaArgument::class.java
//    )
//
//    @JvmField
//    val VALUE_ARGUMENT_NAME: CjPlaceHolderStubElementType<CjValueArgumentName> = CjPlaceHolderStubElementType(
//        "VALUE_ARGUMENT_NAME",
//        CjValueArgumentName::class.java
//    )
//
//    @JvmField
//    val SUPER_TYPE_LIST: CjPlaceHolderStubElementType<CjSuperTypeList> = CjPlaceHolderStubElementType(
//        "SUPER_TYPE_LIST",
//        CjSuperTypeList::class.java
//    )
//
//
//    @JvmField
//    val SUPER_TYPE_CALL_ENTRY: CjPlaceHolderStubElementType<CjSuperTypeCallEntry> = CjPlaceHolderStubElementType(
//        "SUPER_TYPE_CALL_ENTRY",
//        CjSuperTypeCallEntry::class.java
//    )
//
//    @JvmField
//    val SUPER_TYPE_ENTRY: CjPlaceHolderStubElementType<CjSuperTypeEntry> = CjPlaceHolderStubElementType(
//        "SUPER_TYPE_ENTRY",
//        CjSuperTypeEntry::class.java
//    )
//
//    @JvmField
//    val CONSTRUCTOR_CALLEE: CjPlaceHolderStubElementType<CjConstructorCalleeExpression> = CjPlaceHolderStubElementType(
//        "CONSTRUCTOR_CALLEE",
//        CjConstructorCalleeExpression::class.java
//    )
//
//    @JvmField
//    val CONTEXT_RECEIVER: CjContextReceiverElementType = CjContextReceiverElementType("CONTEXT_RECEIVER")
//
//    @JvmField
//    val CONTEXT_RECEIVER_LIST: CjPlaceHolderStubElementType<CjContextReceiverList> = CjPlaceHolderStubElementType(
//        "CONTEXT_RECEIVER_LIST",
//        CjContextReceiverList::class.java
//    )
//
//    @JvmField
//    val NULL: CjConstantExpressionElementType = CjConstantExpressionElementType("NULL")
//
//    @JvmField
//    val BOOLEAN_CONSTANT: CjConstantExpressionElementType = CjConstantExpressionElementType("BOOLEAN_CONSTANT")
//
//    @JvmField
//    val FLOAT_CONSTANT: CjConstantExpressionElementType = CjConstantExpressionElementType("FLOAT_CONSTANT")
//
//    @JvmField
//    val CHARACTER_CONSTANT: CjConstantExpressionElementType = CjConstantExpressionElementType("CHARACTER_CONSTANT")
//
//    @JvmField
//    val INTEGER_CONSTANT: CjConstantExpressionElementType = CjConstantExpressionElementType("INTEGER_CONSTANT")
//
//    //    CjClassLiteralExpressionElementType CLASS_LITERAL_EXPRESSION = new CjClassLiteralExpressionElementType("CLASS_LITERAL_EXPRESSION");
//    @JvmField
//    val COLLECTION_LITERAL_EXPRESSION: CjCollectionLiteralExpressionElementType =
//        CjCollectionLiteralExpressionElementType("COLLECTION_LITERAL_EXPRESSION")
//
//    @JvmField
//    val STRING_TEMPLATE: CjPlaceHolderStubElementType<CjStringTemplateExpression> =
//        CjStringTemplateExpressionElementType("STRING_TEMPLATE")
//
//    @JvmField
//    val LONG_STRING_TEMPLATE_ENTRY: CjPlaceHolderWithTextStubElementType<CjBlockStringTemplateEntry> =
//        CjPlaceHolderWithTextStubElementType(
//            "LONG_STRING_TEMPLATE_ENTRY",
//            CjBlockStringTemplateEntry::class.java
//        )
//
//    @JvmField
//    val SHORT_STRING_TEMPLATE_ENTRY: CjPlaceHolderWithTextStubElementType<CjSimpleNameStringTemplateEntry> =
//        CjPlaceHolderWithTextStubElementType(
//            "SHORT_STRING_TEMPLATE_ENTRY",
//            CjSimpleNameStringTemplateEntry::class.java
//        )
//
//    @JvmField
//    val LITERAL_STRING_TEMPLATE_ENTRY: CjPlaceHolderWithTextStubElementType<CjLiteralStringTemplateEntry> =
//        CjPlaceHolderWithTextStubElementType(
//            "LITERAL_STRING_TEMPLATE_ENTRY",
//            CjLiteralStringTemplateEntry::class.java
//        )
//
//    @JvmField
//    val ESCAPE_STRING_TEMPLATE_ENTRY: CjPlaceHolderWithTextStubElementType<CjEscapeStringTemplateEntry> =
//        CjPlaceHolderWithTextStubElementType(
//            "ESCAPE_STRING_TEMPLATE_ENTRY",
//            CjEscapeStringTemplateEntry::class.java
//        )
//
//
//    @JvmField
//    val UNIT_CONSTANT: CjConstantExpressionElementType = CjConstantExpressionElementType("UNIT_CONSTANT")
//
//    @JvmField
//    val MACRO_EXPRESSION: CjAnnotationEntryElementType = CjAnnotationEntryElementType("MACRO_EXPRESSION")
//
//
//    //    CjPlaceHolderStubElementType<CjAnnotation> ANNOTATION =
//    //            new CjPlaceHolderStubElementType<>("ANNOTATION", CjAnnotation.class);
//    @JvmField
//    val RUNE_CONSTANT: CjConstantExpressionElementType = CjConstantExpressionElementType("RUNE_CONSTANT")
//
//
//    @JvmField
//    val CHARACTER_BYTE_CONSTANT: CjConstantExpressionElementType =
//        CjConstantExpressionElementType("CHARACTER_BYTE_CONSTANT")
//
//
//    @JvmField
//    val INTERFACE: CjInterfaceElementType = CjInterfaceElementType("INTERFACE")
//
//    @JvmField
//
//    val STRUCT: CjStructElementType = CjStructElementType("STRUCT")
//
//
//    @JvmField
//    val MACRO: CjMacroElementType = CjMacroElementType("MACRO")
//
//    @JvmField
//    val FOREIGN: CjForeignDirectiveElementType = CjForeignDirectiveElementType("FOREIGN")
//
//    @JvmField
//    val FOREIGN_BODY: CjPlaceHolderStubElementType<CjForeignBody> = CjPlaceHolderStubElementType(
//        "FOREIGN_BODY",
//        CjForeignBody::class.java
//    )
//
//
//    @JvmField
//    val ENUM: CjEnumElementType = CjEnumElementType("ENUM")
//
//    @JvmField
//    val EXTEND: CjExtendElementType = CjExtendElementType("EXTEND")
//
//    //    CjTupleTypeElementType TUPLE_TYPE = new CjTupleTypeElementType("TUPLE_TYPE");
//    @JvmField
//    val TUPLE_TYPE: CjPlaceHolderStubElementType<CjTupleType> = CjPlaceHolderStubElementType(
//        "TUPLE_TYPE",
//        CjTupleType::class.java
//    )
//
//
//    @JvmField
//    val ENUM_BODY: CjPlaceHolderStubElementType<CjEnumBody> = CjPlaceHolderStubElementType(
//        "ENUM_BODY",
//        CjEnumBody::class.java
//    )
//
//    @JvmField
//    val PROPERTY_BODY: CjPlaceHolderStubElementType<CjPropertyBody> = CjPlaceHolderStubElementType(
//        "PROPERTY_BODY",
//        CjPropertyBody::class.java
//    )
//    @JvmField
//    val PROPERTY_GET: CjPlaceHolderStubElementType<CjPropertyGet> = CjPlaceHolderStubElementType(
//        "PROPERTY_GET",
//        CjPropertyGet::class.java
//    )
//
//    @JvmField
//    val PROPERTY_SET: CjPlaceHolderStubElementType<CjPropertySet> = CjPlaceHolderStubElementType(
//        "PROPERTY_SET",
//        CjPropertySet::class.java
//    )
//
//
//    @JvmField
//    val TYPE_LIST: CjPlaceHolderStubElementType<CjSuperTypeEntry> = CjPlaceHolderStubElementType(
//        "TYPE_LIST",
//        CjSuperTypeEntry::class.java
//    )
//
//
//    @JvmField
//    val TUPLE_LITERAL_EXPRESSION: CjCollectionLiteralExpressionElementType =
//        CjCollectionLiteralExpressionElementType("TUPLE_LITERAL_EXPRESSION")
//
//    @JvmField
//    val IMPORT_DIRECTIVE_ITEM: CjImportDirectiveItemElementType =
//        CjImportDirectiveItemElementType("IMPORT_DIRECTIVE_ITEM")
//
//
//    @JvmField
//    val MAIN_FUNC: CjMainFunctionElementType = CjMainFunctionElementType("MAIN_FUNC")
//    @JvmField
//    val CLASS_INIT: CjClassInitElementType = CjClassInitElementType("CLASS_INIT")
//    @JvmField
//    val CLASS_MAIN_INIT: CjClassInitElementType = CjClassInitElementType("CLASS_MAIN_INIT")
//    @JvmField
//    val CLASS_TILDE_INIT: CjClassInitElementType = CjClassInitElementType("CLASS_TILDE_INIT")
//
//    @JvmField
//    val BASIC_TYPE: CjBasicTypeElementType = CjBasicTypeElementType("BASIC_TYPE")
//
//
//    val CONSTANT_EXPRESSIONS_TYPES: TokenSet = TokenSet.create(
//        BOOLEAN_CONSTANT,
//        FLOAT_CONSTANT,
//        RUNE_CONSTANT,
//        INTEGER_CONSTANT,
//        CHARACTER_CONSTANT,
//
//        REFERENCE_EXPRESSION,
//        DOT_QUALIFIED_EXPRESSION,
//
//        STRING_TEMPLATE,  //            CLASS_LITERAL_EXPRESSION,
//
//        COLLECTION_LITERAL_EXPRESSION
//    )
//
//
//}
